pageserver/src/tenant/mgr.rs

```diff
@@ -306,16 +306,20 @@ pub async fn set_new_tenant_config(
 
 /// Gets the tenant from the in-memory data, erroring if it's absent or is not fitting to the query.
 /// `active_only = true` allows to query only tenants that are ready for operations, erroring on other kinds of tenants.
-pub async fn get_tenant(tenant_id: TenantId, active_only: bool) -> anyhow::Result<Arc<Tenant>> {
+pub async fn get_tenant(
+    tenant_id: TenantId,
+    active_only: bool,
+) -> Result<Arc<Tenant>, TenantStateError> {
     let m = TENANTS.read().await;
     let tenant = m
         .get(&tenant_id)
-        .with_context(|| format!("Tenant {tenant_id} not found in the local state"))?;
+        .ok_or(TenantStateError::NotFound(tenant_id))?;
     if active_only && !tenant.is_active() {
-        anyhow::bail!(
+        tracing::warn!(
             "Tenant {tenant_id} is not active. Current state: {:?}",
             tenant.current_state()
-        )
+        );
+        Err(TenantStateError::NotActive(tenant_id))
```

Comment:
I wonder if here was discussion in earlier versions, but this has now broken our test.

So earlier we used to return an error when the tenant was not active. Now we log and return an error. Is that meant to be like that? Usually we never log and return error; the one who handles the error should log it.

Reply:
which test was broken?
before it was a generic error that the client couldn't know what to do with (Retry/NotRetry/...) here it has a specific error that could be used

Reply:
We fail in general by WARN and ERROR messages. So previously there was no unconditional warn like "Tenant XYZ is not active. Current state: Broken". Now after this got merged there is that.

I don't think that kind of warning is needed or should had been added, it should just had been removed and `Err(TenantStateError::NotActive(tenant_id))` be returned. To re-iterate, previously there was only returning the error with anyhow::bail, as you can see on the diff.

I am just trying to confirm if this logging was intentional as I don't think it was. It would be against our logging/error rfc as well.

Reply:
still don't understand from you which test was broken now

anyway I removed the logging in this PR https://github.com/neondatabase/neon/pull/3834 so plz review

Reply:
The tests added in #3809. For disk usage based eviction, we go through the list of tenants, but only want the active tenants and timelines. In the tests the initial tenant is broken, so it will unconditionally hit this logging statement.